### DISCUSSION TITLE
Are zerocopy and variable length data fully supported?

### DISCUSSION THREAD

**Discussion by xixioba** [Atostek/RustDDS#206] [Q&A]
I see that the Zero-copy receive path feature is done, but I don't know how to test it, it would be nice to have some benchmarks

**Comment by jhelovuo** [comment]
To test zero-copy feature, what would you compare it against?

Maybe you could receive data in your application and then copy it a few times afterwards to see how much time was saved?

If by "variable length data" you mean structures such as strings or arrays, then yes, those are supported.

**Reply by xixioba** [reply]
Thanks, I'll test it.
